src/ipc/handlers/chat_handlers.ts

```diff
@@ -58,6 +62,106 @@ export function registerChatHandlers() {
     return chat.id;
   });
 
+  handle(
+    "chat:create-from-prompt-edit",
+    async (
+      _,
+      params: CreateChatFromPromptEditParams,
+    ): Promise<CreateChatFromPromptEditResult> => {
+      const { appId, chatId, messageId } = params;
+
+      const chat = await db.query.chats.findFirst({
+        where: eq(chats.id, chatId),
+        with: {
+          messages: {
+            orderBy: (messages, { asc }) => [asc(messages.createdAt)],
+          },
+          app: true,
+        },
+      });
+
+      if (!chat) {
+        throw new Error("Chat not found");
+      }
+
+      if (chat.appId !== appId) {
+        throw new Error("Chat does not belong to the provided app");
+      }
+
+      const targetIndex = chat.messages.findIndex(
+        (message) => message.id === messageId,
+      );
+
+      if (targetIndex === -1) {
+        throw new Error("Message not found in chat");
+      }
+
+      const targetMessage = chat.messages[targetIndex];
+      if (targetMessage.role !== "user") {
+        throw new Error("Only user prompts can be edited");
+      }
+
+      let initialCommitHash = chat.initialCommitHash;
+      if (!initialCommitHash) {
+        try {
+          initialCommitHash = await getCurrentCommitHash({
+            path: getDyadAppPath(chat.app.path),
+            ref: "main",
+          });
+        } catch (error) {
+          logger.error("Error getting git revision:", error);
+        }
+      }
+
+      const { newChat, messagesToCopy } = db.transaction((tx) => {
+        const [newChat] = tx
+          .insert(chats)
+          .values({
+            appId,
+            initialCommitHash,
+          })
+          .returning()
+          .all();
+
+        const messagesToCopy = chat.messages.slice(0, targetIndex);
+
+        if (messagesToCopy.length) {
+          tx.insert(messages)
+            .values(
+              messagesToCopy.map((message) => ({
+                chatId: newChat.id,
+                role: message.role,
+                content: message.content,
```

Comment:
can we use spread `{...message}` for all the fields which are verbatim copies? this will make it easier to maintain in the future